### ISSUE TITLE
Lektor eats leading space before dashes

### ISSUE THREAD

**Issue by uranusjr** [lektor/lektor#345]
To Reproduce:

```
body:

Dashes below
 ------
Dashes above
```

Expected output:

```
Dashes below
 ------
Dashes above
```

Real output:

```
Dashes below
------
Dashes above
```

The one space before the dashes disappears. Only exactly one space is stripped, no matter how many you put there (except none, obviously).

**Comment by tariquesani** [comment]
![hello website test](https://cloud.githubusercontent.com/assets/27067/22452658/c662f54a-e79e-11e6-8bf1-d4f43aed79ec.png)

This is what I get if I paste your body text into a Markdown field. Can you elaborate on what is the Type of your body field?

**Comment by uranusjr** [comment]
Oops, I should have mentioned this happens with a markdown field. Did you try the body text *with* a leading space? That one isn’t supposed to produce an `<h2>` tag as in your screenshot. The reason why it does is exactly what I described: Lektor somehow consumes the leading space in the dash line.

Another example to illustrate the problem:

```
body:
Lorem ipsum.

    a code block
    --------
    more text in the code block
```

The above should render to

```
<p>Lorem ipsum.</p>
<pre><code>a code block
--------
more text in the code block
</code></pre>
```

But Lektor gives

```
<p>Lorem ipsum.</p>
<pre><code>a code block</code></pre>
<hr>
<pre><code>more text in the code block</code></pre>
```

Because exactly one leading space before the `--------` line is dropped, breaking indentation.

Minimal reproducible example: https://d.pr/jDob

**Comment by tariquesani** [comment]
Got it! have labeled it as a bug

**Comment by tariquesani** [comment]
I have figured out that it is https://github.com/lektor/lektor/blob/master/lektor/metaformat.py#L88 line which is adding an extra "-" I don't know what can be done to fix it 

@singingwolfboy  Any ideas?

**Comment by uranusjr** [comment]
What I don’t understand about the logic is, why does `_line_is_dashes` strips the line before checking? Shouldn’t it use `rstrip` instead?

**Comment by tariquesani** [comment]
The `_line_is_dashes` logic is fine. What is not fine is assuming any dashed line in content to be `<hr>` and adding a `-` in front of it.

**Comment by uranusjr** [comment]
Well, every line that *starts* with dashes is indeed an `<hr>`. It’s those that start with non-dashes that are not. I asked about the strip logic because it removes leading spaces, and treat (for eaxmple)

    ---

and

        ---

as the same, which is wrong. I didn’t actually try, but it seems replacing `strip` with `rstrip` would probably fix this. Problem is I have no idea if it would break something else.

**Comment by tariquesani** [comment]
Yes, `rstrip` should solve this. I will look at the code a bit more to figure if anything else is affected.
